File: frontend/types/review.ts
export type RiskLevel = 'low' | 'medium' | 'high' | 'critical'

export interface AIReviewStream {
  reviewId: string
  status: 'idle' | 'streaming' | 'complete' | 'error'
  tokens: string[]
  fullText: string
  riskLevel: RiskLevel | null
  issueCount: number | null
}

export interface ParsedReviewBlock {
  type: 'summary' | 'issue' | 'positive' | 'suggestion' | 'risk' | 'heading' | 'text'
  severity?: 'critical' | 'warning' | 'info'
  filename?: string
  line?: number
  text: string
}

export interface StoredReview {
  id: string
  pr_number: number
  repo: string
  full_text: string
  risk_level: RiskLevel
  issue_count: number
  created_at: string
  model_used: string
}

File: backend/app/schemas/review.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class AIReviewRequest(BaseModel):
    repo: str
    pr_number: int
    client_id: str = "default-client"


class AIReviewResponse(BaseModel):
    review_id: str
    status: Literal["streaming", "complete", "error"]


class StoredReview(BaseModel):
    id: str
    pr_number: int
    repo: str
    full_text: str
    risk_level: Literal["low", "medium", "high", "critical"]
    issue_count: int
    created_at: datetime
    model_used: str
